### ISSUE TITLE
Enable Pitch Accents

### ISSUE THREAD

**Issue by ian-battaglia** [lrorpilla/jidoujisho#157]
I put this under "Discussions" a while back since I didn't think it was really an issue, but maybe this is a better place for it.

I just started using jidoujisho, version 2.3.7, and it's really impressive! I'm able to mirror my mining setup on mobile with it. Only issue is that I haven't been able to get the pitch accent graph working.

I'm using a card template that's a slightly older version of [this one](https://github.com/friedrich-de/Basic-Mining-Deck), which adds a pitch accent graph from Kanjium into a field called "Graph." I was looking at the [update from when pitch accents were added, 0.17](https://github.com/lrorpilla/jidoujisho/releases/tag/0.17.4-beta), which is supposed to have a link to the template on how to add the CSS to your cards, but that link is dead. How should I get this to work?

I have two other questions as well, not sure if these should be spun off into their own "issues."

1. How can I enable the ImmersionKit integration mentioned in [2.2](https://github.com/lrorpilla/jidoujisho/releases/tag/2.2)? I'd like to use it to pull sentences and images from if I pull up a word in the dictionary or in a novel, but it's not clear to me how I can enable this.
2. Is it possible to limit which dictionary definitions get added to a card? I have four dictionaries loaded that I like to reference, but usually only want one or two on a card. Is it possible to configure that from the start rather than manually deleting unneeded entries each time?

Thanks!

**Comment by lrorpilla** [comment]
You will need this CSS under your template.

```
.pitch{
  border-top: solid red 2px;
  padding-top: 1px;
}

.pitch_end{
  border-color: red;
  border-right: solid red 2px;
  border-top: solid red 2px;  
  line-height: 1px;
  margin-right: 1px;
  padding-right: 1px;
  padding-top:1px;
}
```
I recommend making a new field for it beside what your current template uses, and using those two fields, one for your current implementation and another for my format. In the future, I will probably make an enhancement that sets the pitch accent to be of other more common formats rather than my own format that I botched up together.

For your other questions,

1. You can enable the ImmersionKit integration by enabling it as an enhancement under User enhancements and then using the new button you'll find after enabling it on the Term field.

2. This is not possible in the current version.

Let me know if you still have any trouble. I am currently at the early development stages of something that I am carefully working on this time while taking much better care of myself (and for the codebase for that matter), so I see myself as in hiatus and not actively looking to make new releases for the current major version at this time, unfortunately.

**Comment by ian-battaglia** [comment]
Awesome, thanks! Got that all to work. I was expecting the ImmersionKit user enhancement to be under "sentence" instead of "term," but that's all set up now. 


Appreciate all your hard work. I'd love in the future to be able to do graphs like these as I think they're a little nicer, but not essential at all.

![Screenshot 2023-02-25 120417](https://user-images.githubusercontent.com/4261596/221372676-54388577-c21a-4bc4-a124-f58c031403a9.jpg)


Finally, is it possible to enable some of [these frequency](https://anacreondjt.gitlab.io/docs/freq/) dictionaries in JidouJisho? I have them added, but haven't seen them pop up. I like to have some way of checking that a word I'm considering adding isn't too uncommon or archaic, and sometimes get a note about the JLPT level, but these would be awesome to add to the dictionary.

Thanks again!

**Comment by lrorpilla** [comment]
HeyThis build includes a work in progress 2.4.0 (only supporting Japanese at the moment) that includes an overhaul of the dictionary system and should be much, much faster than the current system, supports wildcard searching (? or ？ for one character, * or ※ (こめ) for zero to any characters) but will require a fresh installation. You can get the build [here](https://drive.google.com/file/d/1q0drLYf1V7U52yQfmCOoLkR2QL2ou0jO/view?usp=share_link). Let me know if ever you have any more issues.

**Comment by lrorpilla** [comment]
Hey, @ian-battaglia. Missed an edge case which caused issues importing those dictionaries you've linked. I've included the fix for that in a new build below.

This build includes a work in progress 2.4.0 (only supporting Japanese at the moment) that includes an overhaul of the dictionary system and should be much, much faster than the current system, supports wildcard searching (? or ？ for one character, * or ※ (こめ) for zero to any characters) but will require a fresh installation. You can get the build [here](https://drive.google.com/file/d/1q0drLYf1V7U52yQfmCOoLkR2QL2ou0jO/view?usp=share_link). 

As for the graphs, I can probably support them as an enhancement you can enable -- Would you be willing to share the internal structure/how the graphs in your current template work?

**Comment by lrorpilla** [comment]
Changes for the frequency dictionaries have been included in the [**2.4 preview**](https://github.com/lrorpilla/jidoujisho/releases/tag/2.4-preview).

**Comment by ian-battaglia** [comment]
This is great, thanks so much! Going to download and get this running now.

As for the pitch accent graphs, I think they're just natively included as part of the [Kanjium Pitch Accent Dictionary](https://github.com/toasted-nutbread/yomichan-pitch-accent-dictionary). In Yomichan, if you hit the "advanced" settings toggle, there's a few different options for pitch accent graph under the "appearance" tab. Not sure how easy or hard copying this implementation over would be, but it'd definitely be nice to have.

Thanks for all your hard work!

**Comment by lrorpilla** [comment]
For picking out which dictionary definitions get exported, I'm trying to figure out a good user experience for the feature at the moment.

My current idea is to make an auto enhancement for the expanded and collapsed meaning fields that changes the behaviour to "currently expanded" and "currently collapsed" upon opening the Creator or so.

I've had some users think that that should be the default behaviour of the field too, I'd like to implement a way to do it without having the user manually adjust things in the Creator itself, and to allow them to continue to use the app organically as they already would.

**Comment by ian-battaglia** [comment]
> share how your CSS and card format looks

Sure thing. These graphs actually just get dropped directly into the field as it comes in from Yomichan, really not sure how they're rendering it. Here are some screenshots of a sample card I made with Yomichan that has that graph format.

<img width="646" alt="Screenshot 2023-03-03 at 10 51 07 AM" src="https://user-images.githubusercontent.com/4261596/222779462-e8807db8-d9b8-4e72-8f7d-df79f2b2201a.png">
<img width="1387" alt="Screenshot 2023-03-03 at 10 50 52 AM" src="https://user-images.githubusercontent.com/4261596/222779467-32aa5f39-cc40-41a7-b832-fbd5920192a0.png">
<img width="1624" alt="Screenshot 2023-03-03 at 10 50 33 AM" src="https://user-images.githubusercontent.com/4261596/222779468-43141b2d-0799-4e6a-b17f-5db434295dd8.png">
<img width="1624" alt="Screenshot 2023-03-03 at 10 50 28 AM" src="https://user-images.githubusercontent.com/4261596/222779470-1a0bfbfb-d341-4f8e-8c89-8ae18298c660.png">
<img width="1624" alt="Screenshot 2023-03-03 at 10 50 18 AM" src="https://user-images.githubusercontent.com/4261596/222779472-6b82e002-1aa1-45c2-bac2-ac62e1b6ab6e.png">


---

Re: selective definition export, that sounds great. For example, my set up is primarily the 明鏡国語辞典 and 大辞林 dictionary, plus JMdict and KireiCake for slang or whatever. But when I actually make a card, I usually just want the actual dictionary definition from 明鏡国語辞典 (not the part of speech or whatever) and sometimes the first definition or so from JMDict. Right now, I've been just exporting the cards as default, and just going through and editing the definition field in Anki before I review for the day.

**Comment by lrorpilla** [comment]
Hi @ian-battaglia. Just reaching out to let you know selective definition export is available in 2.5. 

I'm not closing this issue yet as I still want to make the pitch accent diagrams template and CSS agnostic, and I'd like to make some modifications to the default template while I'm at it.

**Comment by ian-battaglia** [comment]
Awesome, thanks!
